packages/plugin-layout/src/Layout.ts

```diff
@@ -0,0 +1,111 @@
+import { JWPlugin, JWPluginConfig } from '../../core/src/JWPlugin';
+import {
+    LayoutEngine,
+    Component,
+    componentReference,
+    LayoutEngineConstructor,
+    LayoutEngineConfig,
+} from './LayoutEngine';
+import { VNode } from '../../core/src/VNodes/VNode';
+import { ZoneIdentifier } from './ZoneNode';
+
+let AutomaticPublicId = 0;
+
+interface LayoutMap extends Map<LayoutEngineConstructor, LayoutEngine> {
+    get<A extends LayoutEngineConstructor = LayoutEngineConstructor>(Engine: A): InstanceType<A>;
+}
+
+export class Layout<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T> {
+    readonly engines: LayoutMap = new Map();
+    readonly loaders = {
+        layoutEngines: this.loadLayoutEngines,
+        layoutComponents: this.loadLayoutComponents,
+    };
+    commands = {
+        'UI-show': {
+            title: 'Show a layout component',
+            handler: this.show.bind(this),
+        },
+        'UI-hide': {
+            title: 'Hide a layout component',
+            handler: this.hide.bind(this),
+        },
```

Comment:
Does the commands are intended to be used by a user from a "command bar" or are they intended to be use by the system with execCommand?

Reply:
by buttons, toolbar, close dialog...